scripts/deploy-ballot.ts

```diff
@@ -0,0 +1,55 @@
+import { ethers } from 'hardhat';
+import { Ballot, Ballot__factory } from '../typechain-types';
+import 'dotenv/config';
+
+const { ALCHEMY_API_KEY } = process.env;
+const GOERLI_PRIVATE_KEY = process.env.GOERLI_PRIVATE_KEY as string; // getting linter issues if not assigning type.
+const [networkArg, ...constructorArgs] = process.argv.slice(3);
+
+if (!ALCHEMY_API_KEY) {
+  throw new Error('ALCHEMY_API_KEY missing');
+}
+
+if (!constructorArgs || constructorArgs.length === 0) {
+  throw new Error('You need to pass a valid constructor argument');
+}
+
+const PROPOSALS_BYTES32 = constructorArgs.map(p => ethers.utils.formatBytes32String(p));
```

Comment:
I recommend scoping all of this within the function. global scope variables are usually not desirable

Reply:
Done.